### PR TITLE
Implement ballerina/internal:execBallerinaDoc function

### PR DESCRIPTION
**Pull request by hemikak**

### CODE REVIEW

File: stdlib/ballerina-builtin/src/main/ballerina/internal/exec_ballerina.bal [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2017 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+// Copyright (c) 2017-2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (anupama-pathirage) [inline stdlib/ballerina-builtin/src/main/ballerina/internal/exec_ballerina.bal]: Shall we change this to 2018?

File: stdlib/ballerina-builtin/src/main/ballerina/internal/exec_ballerina.bal [hunk 2]
```diff
@@ -24,4 +24,28 @@ documentation{
...
+public native function execBallerinaDoc(string[] packageList,
+                                        string? sourceRoot = (),
+                                        string? outputPath = (),
+                                        string? templatesPath = (),
+                                        string[]? exclude = (),
+                                        boolean includeNatives = false,
+                                        map<string>? envVars = (),
+                                        string? config = (),
+                                        boolean verbose = false)
+                           returns (string|error);
```
Comment (anupama-pathirage) [inline stdlib/ballerina-builtin/src/main/ballerina/internal/exec_ballerina.bal]: Add a new line?

Comment (hemikak) [reply stdlib/ballerina-builtin/src/main/ballerina/internal/exec_ballerina.bal]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/9227/commits/9b2f2ce167828ea857fabb067b03d97ac5c4f139

File: stdlib/ballerina-builtin/src/main/ballerina/internal/exec_ballerina.bal (hunk 1 above)
Comment (hemikak) [reply stdlib/ballerina-builtin/src/main/ballerina/internal/exec_ballerina.bal]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/9227/commits/9b2f2ce167828ea857fabb067b03d97ac5c4f139

File: stdlib/internal/src/main/java/org/ballerinalang/stdlib/internal/execballerina/ExecBallerinaDoc.java [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+            }
+        }
+    }
+    
+    /**
+     * Build the command to pass to ballerina doc command.
+     * @param context The function context.
+     * @return The command.
+     */
+    private static String buildCommand(Context context) {
```
Comment (manuranga) [inline stdlib/internal/src/main/java/org/ballerinalang/stdlib/internal/execballerina/ExecBallerinaDoc.java]: this function is too long and deep, let's break it up

Comment (hemikak) [reply stdlib/internal/src/main/java/org/ballerinalang/stdlib/internal/execballerina/ExecBallerinaDoc.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/9227/commits/47175543bf31fffd521f8b6d5d61d736aefce8ce

**Pull request merged**